### PR TITLE
Delete leftover configmaps and secrets while cluster deleting

### PR DESCRIPTION
**Pull request by marieroque**
towards https://github.com/giantswarm/giantswarm/issues/30535

That PR is replacing `promtail` occurrences by `logging` to be more generic.

### CODE REVIEW

**Comment by marieroque** [comment]
- [ ] Tested on Vintage
- [ ] Tested on CAPI

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Changed
+
+- Replace promtail occurrences by logging to be more generic.
```
Comment (QuentinBisson) [inline CHANGELOG.md:12]: Why do we want to be more generic?

File: pkg/resource/grafana-agent-config/reconciler.go [hunk 2]
```diff
@@ -99,12 +99,35 @@ func (r *Reconciler) ReconcileCreate(ctx context.Context, lc loggedcluster.Inter
...
 	logger := log.FromContext(ctx)
 	logger.Info("grafana-agent-config delete")
 
-	// Don't do anything, we let observability-bundle do the cleanup when logging.enable=false
+	// Get expected configmap.
+	var currentGrafanaAgentConfig v1.ConfigMap
+	err := r.Client.Get(ctx, types.NamespacedName{Name: getGrafanaAgentConfigName(lc), Namespace: lc.GetAppsNamespace()}, &currentGrafanaAgentConfig)
+	if err != nil {
+		if apimachineryerrors.IsNotFound(err) {
+			logger.Info("grafana-agent-config not found, stop here")
```
Comment (QuentinBisson) [inline pkg/resource/grafana-agent-config/reconciler.go:112]:
```suggestion
			logger.Info("grafana-agent-config not found, skipping")
```

File: pkg/resource/grafana-agent-config/reconciler.go [hunk 3]
```diff
@@ -99,12 +99,35 @@ func (r *Reconciler) ReconcileCreate(ctx context.Context, lc loggedcluster.Inter
...
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, errors.WithStack(err)
+	}
+
+	// Delete configmap.
+	logger.Info("grafana-agent-config deleting", "namespace", currentGrafanaAgentConfig.GetNamespace(), "name", currentGrafanaAgentConfig.GetName())
```
Comment (QuentinBisson) [inline pkg/resource/grafana-agent-config/reconciler.go:119]: You need to remove the finalizer first though

File: pkg/resource/grafana-agent-secret/reconciler.go [hunk 4]
```diff
@@ -113,11 +113,36 @@ func (r *Reconciler) ReconcileCreate(ctx context.Context, lc loggedcluster.Inter
...
+	if err != nil {
+		if apimachineryerrors.IsNotFound(err) {
+			logger.Info("grafana-agent-secret not found, stop here")
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, errors.WithStack(err)
+	}
+
+	// Delete secret.
+	logger.Info("grafana-agent-secret deleting", "namespace", currentGrafanaAgentSecret.GetNamespace(), "name", currentGrafanaAgentSecret.GetName())
```
Comment (QuentinBisson) [inline pkg/resource/grafana-agent-secret/reconciler.go:133]: Same issue about finalizer

File: CHANGELOG.md (hunk 1 above)
Comment (marieroque) [reply CHANGELOG.md:12]: Currently we have a mix of specific (promtail) and generic (logging).
Plus, configmap and secret are naming with a generic name.
Code and logs are more readable IMO with generic naming everywhere :-)

File: pkg/resource/grafana-agent-config/reconciler.go (hunk 3 above)
Comment (marieroque) [reply pkg/resource/grafana-agent-config/reconciler.go:119]: We don't have one

**Pull request merged**